numba/targets/listobj.py

```diff
@@ -158,16 +180,43 @@ def set_dirty(self, val):
         if self._ty.reflected:
             self._payload.dirty = cgutils.true_bit if val else cgutils.false_bit
 
-    def setitem(self, idx, val):
+    def clear_value(self, idx):
+        """Remove the value at the location
+        """
+        self.decref_value(self.getitem(idx))
+        self.zfill(idx, self._builder.add(idx, idx.type(1)))
```

Comment:
Is the zero fill necessary, or just to clear the memory for ease of debugging?

Reply:
it's necessary for the dtor which just decref every slot on it.  Need to add comment here.